frontend/apps/reader/modules/readerlink.lua

```diff
@@ -837,16 +872,33 @@ function ReaderLink:onGoBackLink(show_notification_if_empty)
     end
 end
 
+--- Goes to next location.
+function ReaderLink:onGoForwardLink(show_notification_if_empty)
+    local saved_location = self.location_stack[self.location_stack_index+1]
+    if saved_location then
+        self.location_stack_index = self.location_stack_index + 1
+    end
+    if saved_location then
+        logger.dbg("GoForward: restoring:", saved_location)
+        self.ui:handleEvent(Event:new('RestoreBookLocation', saved_location))
+        return true
+    elseif show_notification_if_empty then
+        UIManager:show(Notification:new{
+            text = _("Location history is empty."),
+        })
+    end
+end
+
 function ReaderLink:onSwipe(arg, ges)
     local direction = BD.flipDirectionIfMirroredUILayout(ges.direction)
     if direction == "east" then
         if isSwipeToGoBackEnabled() then
-            if #self.location_stack > 0 then
+            if self.location_stack_index > 0 or self.location_stack[self.location_stack_index+1] ~= nil then
```

Comment:
What/why is that doing ?
It reads (and reads easier) as `if self.location_stack_index > 0 or self.location_stack[1] ~= nil then`.

Or do you mean `and` instead of `or` ?